htmresearch/algorithms/bayesian_column_pooler_base.py

```diff
@@ -408,10 +418,16 @@ def _activation(weights, input, bias, noise, useBias=True, ignoreNinf=False):
         activation = np.multiply(weights[:, activeMask], input[activeMask]) + noise
         # Special case if active mask has no active inputs (e.g initialisation)
         # then activation becomes 0 and hence the exp of it 1
-        if not ignoreNinf:
-            activation = activation.sum(axis=1) if np.any(activeMask) else activation.sum(axis=1) + np.NINF
-        else:
-            activation = np.ma.masked_invalid(activation).sum(axis=1) if np.any(activeMask) else activation.sum(axis=1) + np.NINF
+        # if not ignoreNinf:
+        #     activation = activation.sum(axis=1) if np.any(activeMask) else activation.sum(axis=1) + np.NINF
+        # else:
+        #     activation = np.ma.masked_invalid(activation).sum(axis=1) if np.any(activeMask) else activation.sum(axis=1) + np.NINF
+
+        # Set explicitly to zero to avoid summing NINF if only one connection is non-existend for a neuron (because multiple objects are activated and only overlapping would be inferred)
```

Comment:
Have to check for equivalence. I would prefer keeping it explicit like this at the moment. 
Avoids bugs and makes it very clear which weights are used.

Also because I am not sure yet about the normalization of input values in the PR.